### ISSUE TITLE
Optional operator has no or negative effect

### ISSUE THREAD

**Issue by stockholmux** [RediSearch/RediSearch#381]
Using the optional query operator (`~` does not perform as documented).

Take this example set:
```
FT.CREATE noodles SCHEMA description TEXT
FT.ADD noodles a82272817-001 1 FIELDS description "To Construct interior alterations to an existing Hong Kong Noodle Express.  (new layout for interior)"
FT.ADD noodles a18226128-011 1 FIELDS description "To construct interior alterations to a General Industrial building production area - Bioneutra (formely Noodle Factory)"
FT.ADD noodles a112817810-002 1 FIELDS description "To change use from Commercial School to  Restaurant (42 seats) and construct interior alterations  (Hung Phat Vietnamese Noodle House)"
FT.ADD noodles a116631333-001 1 FIELDS description "To construct interior alterations to an existing specialty food service - 'Little Asia Noodle House' @ Millbourne Market Mall Food Court."
FT.ADD noodles a182728312-001 1 FIELDS description "to construct interior alterations to an existing restaurant, adding interior partitions, (King's Noodle & Hot Pot Inc)"
FT.ADD noodles a172012770-001 1 FIELDS description "To construct interior alterations to a Restaurant - Pho Hoan Noodle House."
FT.ADD noodles a161975525-001 1 FIELDS description "To construct interior alterations to a restaurant,  \"Oodle Noodle\" Capilano Mall."
FT.ADD noodles a139018595-041 1 FIELDS description "To change use (retail base building to assembly) and  construct interior alterations for a restaurant - Building 8, Newcastle Center HALONG BAY NOODLE HOUSE."
```
Note that each document only has one instance of "noodle". 

Running the query:
```
ft.search noodles "noodle" WITHSCORES
```
The scores are either 0.20000000000000001 or 0.10000000000000001. (Intuitively, I would expect all docs to have the same score, but I can chalk this up to TF-IDF math).

Now, if I run the query:
```
ft.search noodles "noodle ~house" WITHSCORES
```
Only document "a139018595-041" (the first result) gets a score boost (to 1.5), the rest of the scores stay at either 0.20000000000000001 or 0.10000000000000001. Despite that "a112817810-002", "a116631333-001", and "a172012770-001" also contain the term "house".

Note, also I have a situation where I've observed the score going _down_ when using the optional clause but it's in a hard to isolate dataset. I can provide, if needed.

**Comment by dvirsky** [comment]
This has more effect with the DISMAX scorer, and should be used with it.

**Comment by stockholmux** [comment]
@dvirsky Perhaps. Should optional clauses have zero effect on _some_ documents in TF-IDF though? That seems not right.

**Comment by mnunberg** [comment]
Weirdness with optional iterators is solved in 1.6
